pyomo/contrib/derivatives/differentiate.py

```diff
@@ -1,6 +1,7 @@
 from pyomo.core.kernel.component_map import ComponentMap
-import pyomo.core.expr.expr_pyomo5 as _expr
-from pyomo.core.expr.expr_pyomo5 import ExpressionValueVisitor, nonpyomo_leaf_types, value
+import pyomo.core.expr.current as _expr
```

Comment:
why not just import from numeric_expr?

Reply:
No real reason.  I was trying for the minimal set of changes.  Eventually, I would like it so that users outside of `pyomo.core` never have to import anything other than `pyomo.core.expr`.  But that will be a more involved change, with possible problems with backwards compatibility.